File: ProyectoFinal_LP/src/app/Interfaces/Local.ts
// Interface principal para GET
export interface Local {
  id?: number;
  nombre?: string;
  direccion?: string;
  ciudad?: string;
  telefono?: string;
  email?: string;
  capacidad?: number;
  horarioApertura?: string; // formato HH:mm
  horarioCierre?: string; // formato HH:mm
  activo?: boolean;
  createdAt?: Date; // formato ISO DateTime
  updatedAt?: Date; // formato ISO DateTime
}

// Interface para POST (sin id)
export interface LocalPost {
  nombre: string;
  direccion: string;
  ciudad: string;
  telefono: string;
  email: string;
  capacidad: number;
  horarioApertura: string; // formato HH:mm
  horarioCierre: string; // formato HH:mm
  activo: boolean;
}

// Interface para PUT (requiere id)
export interface LocalPut {
  id: number;
  nombre: string;
  direccion: string;
  ciudad: string;
  telefono: string;
  email: string;
  capacidad: number;
  horarioApertura: string; // formato HH:mm
  horarioCierre: string; // formato HH:mm
  activo: boolean;
}

File: ProyectoBackend/pizzeriaBackend/src/main/java/com/pizza/pizzeriaBackend/model/Local.java
package com.pizza.pizzeriaBackend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

@Entity
@Table(name = "locales")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Local {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;
    private String direccion;
    private String ciudad;
    private String telefono;
    private String email;
    private Integer capacidad;
    private LocalTime horarioApertura;
    private LocalTime horarioCierre;
    private boolean activo = true;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "local", cascade = CascadeType.ALL)
    private List<Pedido> pedidos;

    @OneToMany(mappedBy = "local", cascade = CascadeType.ALL)
    private List<Reserva> reservas;

    @OneToMany(mappedBy = "local", cascade = CascadeType.ALL)
    private List<Evento> eventos;
}